### ISSUE TITLE
Power Management - Hybrid Sleep & Battery Critically Low action

### ISSUE THREAD

**Issue by Perkolator** [linuxmint/cinnamon#10154]
```
 * Cinnamon 4.8.6
 * Mint 20.1 (same problem was at least with Mint 20.0, dunno about earlier versions, I'm a new linux user)
 * Nvidia RTX 2060 Mobile, 450.119.03-0ubuntu0.20.04.1
 * 64 bit
 ```

**Issue**

If the `Enable Hybrid Sleep` switch is off in the Power Management settings, system still does HybridSleep when `Battery is critically low` action kicks in, even if `Hibernate` is selected as the action type.

Here's my power settings (the "suspend inactive on a/c" box is empty because I have set custom timeout (and type) with dconf, see the bottom left of the image of those settings):

![Screenshot - hybrid sleep 1](https://user-images.githubusercontent.com/53913724/121677534-a83b7d00-cabe-11eb-821d-5ca05c893deb.png)

And screenshot from journalctl when system tries to do HybridSleep:

![Screenshot - hybrid sleep 2](https://user-images.githubusercontent.com/53913724/121677544-ab366d80-cabe-11eb-8ab9-f651e2b0d4c0.png)

`/etc/UPower/UPower.conf` file has this section:

```
# The action to take when "TimeAction" or "PercentageAction" above has been
# reached for the batteries (UPS or laptop batteries) supplying the computer
#
# Possible values are:
# PowerOff
# Hibernate
# HybridSleep
#
# If HybridSleep isn't available, Hibernate will be used
# If Hibernate isn't available, PowerOff will be used
CriticalPowerAction=HybridSleep
```
If that `CriticalPowerAction` is changed to `Hibernate`, the system does hibernate, not HybridSleep.

My laptop always locks up if the system tries to do HybridSleep. I don't understand why. Suspend alone works, and hibernate alone works too, but not HybridSleep.

***

Is the `/etc/UPower/UPower.conf` file safe from anything changing it, like some system/package update resetting/updating it? Something like `/etc/UPower/UPower.conf.d/10-upower.conf` doesn't seem to exist?

***

Interestingly the `/etc/UPower/UPower.conf` file starts with this comment:
```
# Only the system vendor should modify this file, ordinary users
# should not have to change anything.
```

Here's a wish that the user wouldn't have to edit that file, nor use dconf, i.e. I hope that the Power Management GUI settings would have all the settings (e.g. Time/PercentageLow/Critical/Action et al. settings).

**Comment by mtwebster** [comment]
I don't think we can override those upower settings, but you definitely can override them.

Files in /etc are generally distro/system settings, which means they may want to be updated during package updates. The way debian/ubuntu packaging works for this folder is, if a new version is to be supplied by the update, it compares the *original* old version to the actual current file. If they don't match, you may get a 'file has changed, take a look at the differences' prompt during the update. I believe you can opt to keep the current one or replace it. I think no matter what a backup is made so you can refer back to it for the modifications you made.

I can understand why they prefer to hybrid sleep for this situation, though that's unfortunate it doesn't work for you.

You could maybe try changing `org.cinnamon.settings-daemon.plugins.power percentage-action` to 3 instead of 2 - I would think that would avoid the UPower rule, and let cinnamon-settings-daemon's choice apply instead.

I've made a note to review this stuff for the next version. I don't like these things being hardcoded, but maybe if we tweak some of the default values we can avoid conflicts, and consider adding the action percentage to the gui settings.

**Comment by Perkolator** [comment]
> Files in /etc are generally distro/system settings, which means they may want to be updated during package updates.

That's what I was afraid of. I think I'll have to then create a daily cron job which uses "sed" to compare and replace the settings I want. I can't think of a other way to make sure that my custom settings are safe and work all the time.

> You could maybe try changing `org.cinnamon.settings-daemon.plugins.power percentage-action` to 3 instead of 2 - I would think that would avoid the UPower rule, and let cinnamon-settings-daemon's choice apply instead.

Unfortunately that's another area that's borked too, it doesn't work like you say. I made a post to [here](https://github.com/linuxmint/cinnamon/issues/4367#issuecomment-859561324) about issues regarding to those dconf/gsettings settings. That is actually a 6 year old issue already. Action can not be triggered by those settings. Only through `UPower.conf`. However, like I described there, `UPower.conf` notification settings `PercentageLow / PercentageCritical` doesn't do anything, at least nothing visible to the user, I can't see any notification at all. Notifications work if set through `/org/cinnamon/settings-daemon/plugins/power/` with `percentage-low / percentage-critical` to the same percentages as in `UPower.conf`. The systray power icon still seems to always change to "orangeish" at 20%, no matter what setting user has in `UPower.conf` and/or `/org/cinnamon/settings-daemon/plugins/power/`. It's a big mess unfortunately.

It's really confusing to the user to have an `Enable Hybrid Sleep` setting in GUI and then finding out through logs (after laptop freezes) that the system still tries to do HybridSleep.

***

If you are able to add more GUI settings (perhaps under some "advanced" popup settings view in order to keep it somewhat simple to "regular" users), try to add a delay setting for the `HibernateDelaySec=` to accompany the `Enable Hibernate after suspend` setting. There's an issue for that too [here](https://github.com/linuxmint/cinnamon/issues/9840). Thanks.

**Comment by Perkolator** [comment]
Unfortunately I'm not able to test these things with the latest Mint. Has anyone noticed if this has been fixed/changed?
